impl/maven-core/src/main/java/org/apache/maven/project/DefaultProjectBuilder.java

```diff
@@ -1099,6 +1246,86 @@ public Set<Entry<K, V>> entrySet() {
         }
     }
 
+    /**
+     * Extracts unique module names from the given list of source elements.
+     * A project is considered modular if it has at least one module name.
+     *
+     * @param sources list of source elements from the build
+     * @return set of non-blank module names
+     */
+    private Set<String> extractModules(List<org.apache.maven.api.model.Source> sources) {
```

Comment:
Method could be static.

Reply:
Done in 0c2451e7a30f8f427347b6c8b10bca3f551db02d. Made extractModules method static.